src/screenbot.js

```diff
@@ -24,20 +30,56 @@ var Screenbot = (function Screenbot() {
              channel_token;
     };
 
-    var _request = function(endpoint, callback){
+    var _parseResponse = function(responseText){
+      try {
+        var response = JSON.parse(responseText);
+      }
+      catch (e) {
+        return _handleError({ code: ERROR_CODES.PARSE_ERROR.code });
+      }
+
+      return response;
+    };
+
+    var _request = function(endpoint, cb){
       var xhttp = new XMLHttpRequest();
       xhttp.onreadystatechange = function() {
         if (xhttp.readyState === 4 && xhttp.status === 200) {
-          var response = JSON.parse(xhttp.responseText);
-          callback(response.ok, response);
+          var response = _parseResponse(xhttp.responseText)
+          if(response instanceof Error || !response.ok) cb(response);
+
+          cb(null, response);
         } else if(xhttp.readyState === 4) {
-          callback(false, xhttp.responseText);
+          cb(_parseResponse(xhttp.responseText));
         }
       };
       xhttp.open("GET", endpoint, true);
       xhttp.send();
     };
 
+    var _handleError = function(args){
+      return new ScreenbotError(args.code, args.message);
+    };
+
+    var ScreenbotError = function(code, message){
```

Comment:
I realize this is kind of counter to my previous comment but seeing more how this implemented makes me think that the options as the first argument and then changing the second argument to optionally accept what could be called the `originalException`. Placing the `originalException` as a member on the object would give the original stack trace and message from the error if needed.